dask/utils.py

```diff
@@ -924,3 +924,73 @@ def natural_sort_key(s):
     """
     return [int(part) if part.isdigit() else part
             for part in re.split('(\d+)', s)]
+
+
+def factors(n):
+    """ Return the factors of an integer
+
+    https://stackoverflow.com/a/6800214/616616
+    """
+    seq = ([i, n // i] for i in range(1, int(pow(n, 0.5) + 1)) if n % i == 0)
+    return set(functools.reduce(list.__add__, seq))
```

Comment:
nit: this version looks a little more idiomatic to me: `set(elem for pair in seq for elem in pair)`

Reply:
I agree.  I'm just trying to keep this as close to the StackOverflow version as possible while still satisfying the style checker.  I'm inclined to externally sourced code as-is.